central/imagev2/datastore/store/postgres/store.go

```diff
@@ -585,7 +592,11 @@ func (s *storeImpl) retryableGet(ctx context.Context, id string) (*storage.Image
 	if err != nil {
 		return nil, false, err
 	}
-	image, found, err := s.getFullImage(ctx, tx, id)
+	// Add tx to the context to ensure image metadata plus its components and CVEs are all retrieved
+	// in the same transaction as the updates.
+	ctx = postgres.ContextWithTx(ctx, tx)
+
+	image, found, err := s.getFullImage(ctx, id)
```

Comment:
We are losing any err returned from `getFullImage`.  We probably should report that `err`, but we also need to make sure in any of these that we close the transaction.

Take a look at https://github.com/stackrox/stackrox/blob/master/central/blob/datastore/store/store.go for examples